src/frontend/parser.cpp

```diff
@@ -26,18 +26,27 @@
 #include "frontend/parser.h"
 #include "frontend/expr.h"
 #include "frontend/symbol.h"
+#include "frontend/diagnostic.h"
 #include "runtime/value.h"
 #include "location.h"
 
 //#define TRACE(x) do { fprintf(stderr, "%s\n", x); } while (0)
 #define TRACE(x) do { } while (0)
 
+extern DiagnosticReporter *reporter;
```

Comment:
Does a simple `extern DiagnosticReporter` work?

Reply:
You can't create an object of an abstract class.